packages/next/src/server/normalizers/built/app/app-page-normalizer.ts

```diff
@@ -1,12 +1,35 @@
 import { PAGE_TYPES } from '../../../../lib/page-types'
 import { AbsoluteFilenameNormalizer } from '../../absolute-filename-normalizer'
+import { Normalizers } from '../../normalizers'
 
 /**
  * DevAppPageNormalizer is a normalizer that is used to normalize a pathname
  * to a page in the `app` directory.
  */
-export class DevAppPageNormalizer extends AbsoluteFilenameNormalizer {
+class DevAppPageNormalizerInternal extends AbsoluteFilenameNormalizer {
   constructor(appDir: string, extensions: ReadonlyArray<string>) {
     super(appDir, extensions, PAGE_TYPES.APP)
   }
 }
+
+export class DevAppPageNormalizer extends Normalizers {
+  constructor(
+    appDir: string,
+    extensions: ReadonlyArray<string>,
+    _isTurbopack: boolean
+  ) {
+    const normalizer = new DevAppPageNormalizerInternal(appDir, extensions)
+    super(
+      // %5F to _ replacement should only happen with Turbopack.
+      // TODO: enable when page matcher `/_` check is moved: https://github.com/vercel/next.js/blob/8eda00bf5999e43e8f0211bd72c981d5ce292e8b/packages/next/src/server/route-matcher-providers/dev/dev-app-route-route-matcher-provider.ts#L48
```

Comment:
Added the placeholder for what the solution should be but we can't enable it yet because of the `/_` check after calling the page normalizer.